ghcide/src/Development/IDE/Core/Compile.hs

```diff
@@ -56,9 +55,9 @@ import           HieDb
 import           Language.LSP.Types                (DiagnosticTag (..))
 
 #if MIN_VERSION_ghc(8,10,0)
```

Comment:
There's quite a lot of CPP in this module, can we hide any more of it in the `Compat` modules?

Reply:
This would be great, but I don't have any more time to work on it. Volunteers are welcome

Reply:
This one was opened as a report https://github.com/haskell/haskell-language-server/issues/2568 in the mainline.